packages/admin/admin/src/FinalFormSaveSplitButton.tsx

```diff
@@ -10,9 +10,10 @@ import { useStackApi } from "./stack/Api";
 
 export interface FormSaveButtonProps {
     localStorageKey?: string;
+    onNavigateToEditPage?: () => void;
```

Comment:
In the `FinalForm` component we call this callback `onAfterSubmit`. I think it would be good to use the same name for both components

Reply:
that one is different: it is only called when Save (not Save and go back) is clicked.

Reply:
Okay, but isn't onNavigateToEditPage to specific then? Could there be other use cases? I can think of sending an analytics event after a successful save

Reply:
That would't belong into the submit button.

it *is* a very specific event.... `onAfterSubmitNotGoingBack`?

(I don't like the whole implementation altogether)

Reply:
> (I don't like the whole implementation altogether)

Me neither. The FinalFormSaveSplitButton is very inflexible. For instance, it is not possible to disable the "Save and go back"-button

could we change the callback to work similarly to the FinalForm callback? So that `stackApi.goBack()` is the default behavior, but can be replaced with `onAfterSubmit`: https://github.com/vivid-planet/comet/blob/d79739ba1fba550ff53c8e0ed745079e7e997d60/packages/admin/admin/src/FinalForm.tsx#L27-L31

The callback would then be `onAfterSubmit?: (goBack: boolean) => void`. I'm not sure if this is a better approach.

Reply:
> Me neither. The FinalFormSaveSplitButton is very inflexible. For instance, it is not possible to disable the "Save and go back"-button

This component is for a very specific use case, it disables itself based on form state. Additional disabling or only disabling one of the two buttons using props is not something I would see useful. Other props might be useful, but where to start? maybe add a `buttonProps` prop that is appplied to all buttons?

Reply:
> could we change the callback to work similarly to the FinalForm callback? So that stackApi.goBack() is the default behavior

the default we (usually) can keep, we need to redirect only after save (without going back) for creation.

Reply:
> Other props might be useful, but where to start?

I've just search across all our projects, the component hasn't been used in any project yet. Maybe we should wait until it's actually used and then adapt accordingly.

Reply:
Let's keep the name as it is. For now, the only use case for this callback we can think of is to navigate to the edit page. We may rename it later if it is used for other cases in the future.